### PR TITLE
Project folder restructure, added Token generation from backend

### PR DESCRIPTION
**Pull request by mrvanagas**
Restructured project folder to separate client and server folders. Added server to generate a token. At the moment, token is generated by user click of a button as show in the picture:
image

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -20,6 +20,8 @@
   "dependencies": {
     "axios": "0.21.4",
     "core-js": "3.2.1",
+    "dotenv": "^16.0.0",
+    "dotenv-webpack": "^7.1.0",
```
Comment (JuliaShavyrova) [inline client/package.json:23]: do you need both packages?

File: client/src/API/api.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  console.log(token);
+};
+
+const weatherAPI = axios.create({
+  baseURL: 'https://pfa.foreca.com/api/v1/',
+  headers: { authorization: `Bearer ${token}` }
+});
+
+const lon = '23.9036';
+const lat = '54.8985';
```
Comment (JuliaShavyrova) [inline client/src/API/api.js]: no need to store fixed position

File: client/src/Utils/geoLocator.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const successCallback = position => {
+  console.log(position);
+};
+
+const errorCallback = error => {
+  console.error(error);
+};
+
+navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
```
Comment (JuliaShavyrova) [inline client/src/Utils/geoLocator.js]: don't see any `export` in the file, but it actually imported in `App` component. just interested - does it work?

File: client/src/configs/token.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+const token =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9wZmEuZm9yZWNhLmNvbVwvYXV0aG9yaXplXC90b2tlbiIsImlhdCI6MTY1MDYzNjIwMiwiZXhwIjo5OTk5OTk5OTk5LCJuYmYiOjE2NTA2MzYyMDIsImp0aSI6ImU3ODNiOTA5NDdhZjZlYjciLCJzdWIiOiJsdWtvbmVsaXMxIiwiZm10IjoiWERjT2hqQzQwK0FMamxZVHRqYk9pQT09In0.aZP7m278J_ToDZFqLUVBARHs161WQf-bjhbwa1y_fIo';
```
Comment (JuliaShavyrova) [inline client/src/configs/token.js]: please don't store hardcoded token in repository

File: client/src/Utils/geoLocator.js (hunk 3 above)
Comment (mrvanagas) [reply client/src/Utils/geoLocator.js]: Yes, GeoLocation is called automatically once the app is loaded:
image
We can see getLocation is not called in App.jsx, however in browser console we can see that it has been called automatically without any user input:
image
Why does it work like that, I have not figured out exactly.

File: client/src/API/api.js (hunk 2 above)
Comment (mrvanagas) [reply client/src/API/api.js]: yes, this is for testing purposes. The idea for the first iteration is for the app to automatically get the user location and show the weather info according to the coordinates.

File: client/package.json (hunk 1 above)
Comment (mrvanagas) [reply client/package.json:23]: No, one of them will be removed, once I figure out which one I need.

File: client/src/configs/token.js (hunk 4 above)
Comment (mrvanagas) [reply client/src/configs/token.js]: This will be removed, once I implement silent authorization.

File: client/src/API/api.js (hunk 2 above)
Comment (JuliaShavyrova) [reply client/src/API/api.js]: please make it work without hardcoded values, it's much easier to do it now then later with more code added

File: client/src/Utils/geoLocator.js (hunk 3 above)
Comment (JuliaShavyrova) [reply client/src/Utils/geoLocator.js]: > Why does it work like that, I have not figured out exactly.

please make your code more predictable or find out why it works that way and can be used here :)

File: client/src/API/api.js (hunk 2 above)
Comment (mrvanagas) [reply client/src/API/api.js]: Fixed hard coded values, now coordinates are given from getLocation() function

File: client/src/Utils/geoLocator.js (hunk 3 above)
Comment (mrvanagas) [reply client/src/Utils/geoLocator.js]: Reworked the geoLocator.js file, now it activates only when called and it is being called in the API.js file, in order to supply current coordinates of the user to the app.

EDIT: The promise does not pass the coordinate values to the coord variables, investigating now.

File: client/package.json (hunk 1 above)
Comment (mrvanagas) [reply client/package.json:23]: Got rid of dotenv dependancy

File: client/src/Utils/geoLocator.js (hunk 3 above)
Comment (mrvanagas) [reply client/src/Utils/geoLocator.js]: Reworked the logic of geolocator.js file and used hooks to call and use the information.

File: client/src/Utils/geoLocator.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// export default async function getLocation() {
+//   return new Promise((resolve, reject) => {
+//     navigator.geolocation.getCurrentPosition(resolve, reject);
+//   });
+// }
+
+//original location function
```
Comment (JuliaShavyrova) [inline client/src/Utils/geoLocator.js]: please remove unused code

File: client/src/configs/token.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const token =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9wZmEuZm9yZWNhLmNvbVwvYXBpXC92MVwvXC9hdXRob3JpemVcL3Rva2VuIiwiaWF0IjoxNjUxMjE0NTgxLCJleHAiOjk5OTk5OTk5OTksIm5iZiI6MTY1MTIxNDU4MSwianRpIjoiMTI0ZWNlOTg5MGY5ZDRiOCIsInN1YiI6Imx1a29uZWxpczEiLCJmbXQiOiJYRGNPaGpDNDArQUxqbFlUdGpiT2lBPT0ifQ.yrTZFMKWGHro_ZPBIBRgQ1E8i36quPRZb4NJDCkRkx8';
```
Comment (JuliaShavyrova) [inline client/src/configs/token.js]: still see hardcoded token, please remove it

Comment (mrvanagas) [reply client/src/configs/token.js]: hard coded token removed

File: client/src/Utils/geoLocator.js (hunk 5 above)
Comment (mrvanagas) [reply client/src/Utils/geoLocator.js]: Removed unused code

File: client/src/configs/token.js (hunk 6 above)
Comment (JuliaShavyrova) [reply client/src/configs/token.js]: Do you still need this file?

File: client/src/configs/token.js (hunk 4 above)
Comment (mrvanagas) [reply client/src/configs/token.js]: Silent auth implemented, removed token file

**Review by JuliaShavyrova — APPROVED** [review]

**Comment by JuliaShavyrova** [comment]
please check destination branch

**Comment by mrvanagas** [comment]
Done, changed destination branch.

**Pull request merged**